### PR TITLE
Prop-types + dependency updates

### PR DESCRIPTION
**Pull request by JulianKingman**
I created this PR to switch from React.PropTypes to the `prop-types` library for proptype validation (needed in RN 0.44+). One package I couldn't update is the shoutem fork of lightbox (https://github.com/shoutem/react-native-lightbox), which still uses proptypes (I think the original package has been updated to remove proptypes, and there's an outstanding PR on `react-native-view-transformer` to switch it over https://github.com/ldn0x7dc/react-native-view-transformer/issues/13).

### CODE REVIEW

**Comment by JulianKingman** [comment]
Same goes for transformable image as lightbox: https://github.com/ldn0x7dc/react-native-transformable-image/pull/30

**Comment by JulianKingman** [comment]
Also, `createClass` in lightbox has to go (https://github.com/shoutem/react-native-lightbox/blob/master/LightboxOverlay.js)

And the navigation experimental package needs an update: https://github.com/goshakkk/react-native-navigation-experimental-compat/issues/1

**Comment by bm-software** [comment]
@shoutem-owner Please merge this PR and release...I need this

**Comment by tquiroga** [comment]
Blocked completely on RN 0.49 with React 16 as it's been removed completely. Merge would be great.

**Comment by ifsnow** [comment]
Do you have any plans about this? I hope to be as soon as possible.

**Comment by Firanus** [comment]
I'd would also love to see this merged and released soon guys.

**Comment by CrazyPython** [comment]
`npm install @shoutem/ui#242/head\#prop-types`

**Comment by CrazyPython** [comment]
This PR fixes the PropTypes issue for shoutem/ui, but shoutem/theme and shoutem/animation as on NPM are broken. Their github repo has the fix, but they haven't published it. Not only that, they're relying on a version of react-native-view-transformer that also has the PropTypes issue. It seems like Shoutem UI has been abandoned.

Time to use something like https://github.com/robinpowered/glamorous-native.

**Comment by zrubenst** [comment]
Can this be worked on?

**Comment by Robophil** [comment]
Hi guys. I'm giving `shoutem` a test and ran into this.
Can this be merged in soonest?

**Comment by cihadturhan** [comment]
@tquiroga same here. I wanted to use this with [infinitered/ignite/](/infinitered/ignite/) and I won't have an option to wait. Probably I'll make a pr from this to my branch and move ahead.

**Comment by CrazyPython** [comment]
Look, it's been over two months since this pull request was sent. Despite many of us begging shoutem to merge this, they haven't even posted a comment. This project is abandoned. That's the sad truth.

**Comment by CrazyPython** [comment]
@zrumenjak at least give us an update please?

**Comment by dodsky** [comment]
Hi everyone,

first of all, we want to thank each of you for giving back so much and we want to apologize for all the headaches we caused you in the last few months. It was really stressful for us too. We are aware that this library didn't get love and care it deserves, for that we are truly sorry. 

TLDR:
We will start maintaining this library again as of next week, and I will be taking over the efforts, along with help of previous maintainers (@zrumenjak @SoHotSoup) and you guys - the contributors. 

Not-so-TLDR:
As you all know, this library was created as a tool to "supercharge" your React Native development and is a part of a broader Shoutem development platform for developing React Native apps. We wanted to give everyone the tools to create and maintain modular, great lookin', and performant mobile apps. The same tools that we use in day to day development as a software agency. However, we were super busy doing other things lately...  Instead of working on the platform and libraries, we were busy trying to "fight" Apple's latest app guidelines, which effectively crushed Shoutem's business model (selling affordable apps to SMBs). We tried to contact and explain what we are ultimately trying to build here, and we wanted to know how all the new rules will affect us. Apple, being the juggernaut it is, completely ignored all our efforts. In short, Apple's new rules made us stop and rethink our strategy, and forced us to take new directions, as a team, and as individuals... A lot of people are switching to new opportunities (including @zrumenjak and @SoHotSoup), and we are now running in maintenance mode. All maintenance efforts will be driven by pure enthusiasm and love from us, the developers that created all this (mostly in our free time) so bear with us... We'll also put our efforts into cleaning up and open sourcing the whole platform so that anyone can use it.

Thank you for reading. Best,
@dodsky

**Comment by JulianKingman** [comment]
It looks like most of these are already implemented, at this point it just looks like a couple dependency updates. Do you want to merge this, or just make the package updates yourself and have me close this?

**Comment by dodsky** [comment]
Hi @JulianKingman, we had a few of PRs with similar changes so we opted to create a clean branch and fix everything in one place, so you can close this one I. Changes that were implemented in this PR too is published in version `0.22.0`.
